lib/queueClient.ts

```diff
@@ -0,0 +1,209 @@
+    super(name, context);
+    if (!options) options = {};
+    this.receiveMode = options.receiveMode || ReceiveMode.peekLock;
+    this.maxConcurrentCalls = options.maxConcurrentCalls != undefined ? options.maxConcurrentCalls : 1;
+  }
+
+  /**
+   * Closes the AMQP connection to the ServiceBus Queue for this client,
+   * returning a promise that will be resolved when disconnection is completed.
+   * @returns {Promise<any>}
+   */
+  async close(): Promise<any> {
+    try {
+      if (this._context.namespace.connection && this._context.namespace.connection.isOpen()) {
+        // Close the sender.
+        if (this._context.sender) {
+          await this._context.sender.close();
+        }
+        // Close the receiver.
+        if (this._context.streamingReceiver) {
+          await this._context.streamingReceiver.close();
+        }
+        // Close the management session
+        await this._context.managementSession!.close();
+        debug("Closed the client '%s'.", this.id);
+      }
+    } catch (err) {
+      const msg = `An error occurred while closing the queue client ` +
+        `"${this.id}": ${JSON.stringify(err)} `;
+      debug(msg);
+      throw new Error(msg);
+    }
+  }
+
+  /**
+   * Sends the given message to the ServiceBus Queue.
+   *
+   * @param {any} data  Message to send.  Will be sent as UTF8-encoded JSON string.
+   * @returns {Promise<Delivery>} Promise<Delivery>
+   */
+  async send(data: SBMessage): Promise<Delivery> {
+    const sender = MessageSender.create(this._context);
+    return await sender.send(data);
+  }
+
+  /**
+   * Send a batch of Message to the ServiceBus Queue. The "message_annotations", "application_properties"
+   * and "properties" of the first message will be set as that of the envelope (batch message).
+   *
+   * @param {Array<Message>} datas  An array of Message objects to be sent in a Batch
+   * message.
+   *
+   * @return {Promise<Delivery>} Promise<Delivery>
+   */
+  async sendBatch(datas: SBMessage[]): Promise<Delivery> {
+    const sender = MessageSender.create(this._context);
+    return await sender.sendBatch(datas);
+  }
+
+  /**
+   * Starts the receiver by establishing an AMQP session and an AMQP receiver link on the session.
+   * Messages will be passed to the provided onMessage handler and error will be passed to the
+   * provided onError handler.
+   *
+   * @param {OnMessage} onMessage          The message handler to receive Message objects.
+   * @param {OnError} onError              The error handler to receive an error that occurs
+   * while receiving messages.
+   *
+   * @returns {ReceiveHandler} ReceiveHandler - An object that provides a mechanism to stop
+   * receiving more messages.
+   */
+  receive(onMessage: OnMessage, onError: OnError, options?: MessageHandlerOptions): ReceiveHandler {
+    if (!this._context.streamingReceiver ||
+      (this._context.streamingReceiver && !this._context.streamingReceiver.isOpen())) {
```

Comment:
the second check of `this._context.streamingReceiver` isn't needed. the full condition can be:

```ts
if (!this._context.streamingReceiver || !this._context.streamingReceiver.isOpen()) {
```